### PR TITLE
Add string for Leaf2 CN

### PR DESCRIPTION
**Pull request by AndyCGYan**
Verified by decrypting an upx.

### CODE REVIEW

**Comment by crackself** [comment]
Hi, my Leaf2 aslo have a modle "Leaf2_P", maybe related to product in China homeland
Screenshot_20221014-180535

**Comment by Hagb** [comment]
> Hi, my Leaf2 aslo have a modle "Leaf2_P", maybe related to product in China homeland!

There are two models named "Leaf2". The `MODEL` string of his one is `Leaf2_P`, and `MODEL` string of the other one is `Leaf2`(#47). I don't know the difference of these two Leaf2.

Using the method in https://github.com/Hagb/decryptBooxUpdateUpx/issues/2#issuecomment-704006389 to get the update information of these two variants, I found that:

- Information in English of `Leaf2` exists: <http://data.onyx-international.cn/api/firmware/update?where={"buildNumber":0,"buildType":"user","deviceMAC":"","lang":"en_US","model":"Leaf2","submodel":"","fingerprint":""}>
- Information in Chinese of `Leaf2` doesn't exist: <http://data.onyx-international.cn/api/firmware/update?where={"buildNumber":0,"buildType":"user","deviceMAC":"","lang":"zh_CN","model":"Leaf2","submodel":"","fingerprint":""}>
- Information in English of `Leaf2_P` not exists: <http://data.onyx-international.cn/api/firmware/update?where={"buildNumber":0,"buildType":"user","deviceMAC":"","lang":"en_US","model":"Leaf2_P","submodel":"","fingerprint":""}>
- Information in English of `Leaf2_P` not exists: <http://data.onyx-international.cn/api/firmware/update?where={"buildNumber":0,"buildType":"user","deviceMAC":"","lang":"zh_CN","model":"Leaf2_P","submodel":"","fingerprint":""}>

Probably just as @crackself said, `Leaf2_P` is a China Mainland variant.

**Comment by Hagb** [comment]
Could anyone else has a Leaf2 tell us

- Whether your Leaf2 was bought in China Mainland?
- What is the `MODEL` string of your Leaf2? (You can use apk from https://github.com/Hagb/GetBooxUpxKeysApp/releases or run command `adb shell getprop ro.product.model` to get the `MODEL` string.)

**Comment by sargunv** [comment]
Purchased in USA, my model is "Leaf2"

**Comment by CrazyCoder** [comment]
I purchased 2 versions of Leaf2: Back and White. Both were bought from the official store at boox.com.
The white version was shipped from China: Hong Kong / White Edition (To Be Shipped Around Nov 6).
Black version was shipped from the US warehouse: US / Black Edition.
Both have the same keys: https://github.com/Hagb/decryptBooxUpdateUpx/pull/47.

**Comment by AndyCGYan** [comment]
My `Leaf2_P` was purchased in mainland CN on release day.
BTW, mainland CN doesn't sell a white variant (yet), so the above HK white could well be commonized with INTL.

**Comment by RADM90** [comment]
@Hagb  Please find below strings and fingerprint for Leaf2 (white) which is officially imported in South Korea.
(To avoid any mistypo or confusion, attached `upx key extraction` result screenshot.)
```
"MODEL" = "Leaf2",
"STRING_SETTINGS" = "okRZqgfOHft/Y5jHx1N0ECjqrc/Op/nADFj2uL7Jbuxq",
"STRING _UPGRADE" = "10Na33u5HfsCVHqjtwLrwACOW+2cz5MSnXUogONCTd9c"
# fingerprint: Onyx/Leaf2/Leaf2:11/2022-11-22_10-20_3.3.1_e4f27d41a/1824:user/dev-keys
```
Screenshot_20230305-014056


Also please be noted I am wandering with below unicode decoding exception.

### Description
- As per the newest firmware information from the API presents `"updatedAt":"2022-10-31T10:18:51.093Z"`, I used `update.upx` of `2022-11-14_16-04_3.3.1_41f225958` which is the most recent one I could get.
- Inserted below JSON string on the line 208 of `DeBooxUpx.py` according to the instruction as you mentioned.
    ```
    'Leaf2': {
        "MODEL": "Leaf2",
        "STRING_SETTINGS": "okRZqgfOHft/Y5jHx1NOECjqrc/Op/nADFj2uL7Jbuxq",
        "STRING_UPGRADE": "10Na33u5HfsCVHqjtwLrwACOW+2cz5MSnXUogONCTd9c"
    }
    ```
- While executing the command `python DeBooxUpx.py Leaf2`, I faced following `UnicodeDecodeError`
    ```
Traceback (most recent call last):
  File "DeBooxUpx.py", line 274, in <module>
    decrypter = DeBooxUpx(**boox_strings[device_name])
  File "DeBooxUpx.py", line 226, in __init__
    self.decryptStr(tmpKey, STRING_SETTINGS))
  File "DeBooxUpx.py", line 233, in decryptStr        
    return cipher.decrypt(b64decode(string)).decode().strip()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfb in position 16: invalid start byte
    ```

Encryption and Hashing is out of my knowledge but something what I've got is, it happened with `/` on my `STRING_SETTINGS`.
- Decrypted binary code with Base64 is `b'4131D58DC6CB5D\x025\xfbn\x98\xf3\xaeg\x0f>576CAB16\n` in this case.
- Before feeding `UPX Steram`, working good with usual MD5 instead of my `"STRING_SETTINGS"`

**Review by Hagb — CHANGES_REQUESTED** [review]
According to the above information, could you please rename this `Leaf2` to `Leaf2-cn` and add information about this `-cn` suffix? (Just like `-ru` suffix)

**Comment by AndyCGYan** [comment]
> ...add explanation of this `-cn` suffix? (Just like `-ru` suffix)

Just noticed the explanation part, will append later.

File: README.md [hunk 1]
```diff
@@ -110,7 +111,7 @@ decrypter.deUpx(updateUpxPath, decryptedPath)
 |livingstone-ru|`LIVINGSTONE`|`ES5zbiHTesxFb+zdkjxiqJ+1dOwyOCv2BzCV7fOYDJxf`|`EFwGaFSherxX1k6Hl/U6TiQhGJOzTPDfsPzHg+z9guln`|`fQ5TLnfKUOHq0f7XXdu9b1FGAckqI576ZBPkZfPg`|
 <!--(strings table end)-->
 
-The models sold in Russia are marked with suffix `-ru`, **some or all of the firmwares for which don't use the same strings with their international versions!**
+The models sold in China/Russia are marked with suffixes `-cn`/`-ru`, **some or all of the firmwares for which don't use the same strings with their international versions!**
```
Comment (Hagb) [inline README.md:114]: The explanation may be a bit confusing. As far as I known, most (if not all) of models sold Russia have different firmwares with their international versions, but only a few Chinese models has different firmwares with international ones.

**Review by Hagb — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (AndyCGYan) [reply README.md:114]: > The explanation may be a bit confusing. As far as I known, most (if not all) of models sold in Russia have different firmwares with their international versions, but only a few Chinese models has different firmwares with international ones.

Perhaps just tell me what you think sounds best? I'm not familiar with the Boox ecosystem, the way I wrote it is more to reduce clutter.

Comment (Hagb) [reply README.md:114]: > > The explanation may be a bit confusing. As far as I known, most (if not all) of models sold in Russia have different firmwares with their international versions, but only a few Chinese models has different firmwares with international ones.
> 
> Perhaps just tell me what you think sounds best? I'm not familiar with the Boox ecosystem, the way I wrote it is more to reduce clutter.

```suggestion
A few of models sold in China and most (if not all) of models sold in Russia are marked with suffixes `-cn`/`-ru`, **the firmwares for which don't use the same strings with their international versions!**
```
How about this?

**Comment by Hagb** [comment]
Thank you and everyone who provided the information!

**Pull request merged**